Fix `Build and Deploy` workflow

After merging #455, the `Build and Deploy` workflow fails, see:
https://github.com/carpentries-incubator/deep-learning-intro/actions/runs/8881072163/job/24382959500

```shell
[...]
── Creating Images page ────────────────────────────────────────────────────────
  Writing ''instructor/images.html''
  Writing ''images.html''
── Creating Instructor Notes ───────────────────────────────────────────────────
  Error: Error: Length of `path` must equal length of `new_path`
  Running git worktree remove --force \
    /home/runner/work/deep-learning-intro/deep-learning-intro/site/docs
  Running git worktree remove --force \
    /home/runner/work/deep-learning-intro/deep-learning-intro/site/built
  Execution halted
  Error: Process completed with exit code 1.
  ##[debug]Finishing: Deploy Site
```

I cannot reproduce this locally; it passes fine:
```shell
% R -e 'sandpaper::serve(quiet = FALSE, port = "3435")'

R version 4.3.2 (2023-10-31) -- "Eye Holes"
[...]

── Creating Images page ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Writing 'instructor/images.html'
Writing 'images.html'
── Creating Instructor Notes ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Writing 'instructor/instructor-notes.html'
Writing 'instructor-notes.html'
── Creating sitemap.xml ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Output created: /Users/carstenschnober/Courses/teaching/deep-learning-intro/site/docs/index.html
Serving the directory /Users/carstenschnober/Courses/teaching/deep-learning-intro/site/docs at http://127.0.0.1:3435

```

--- a/config.yaml
+++ b/config.yaml
@@ -71,7 +71,6 @@ learners:
 
 # Information for Instructors
 instructors:
-- schedule.md
 
 # Learner Profiles
 profiles:

